openhcl/page_pool_alloc/src/lib.rs

```diff
@@ -47,27 +48,38 @@ enum State {
         pfn_bias: u64,
         #[inspect(hex)]
         size_pages: u64,
+        device_id: u64,
+        device_name: String,
```

Comment:
This is quite expensive. Can you keep an index back into some table instead?

Reply:
You could also use `Arc<str>`, but that's still expensive.

Reply:
Yeah i'm refactoring this given our discussion on the IDs - I'll probably index into a string table.